### ISSUE TITLE
weak_delegate rule false positive

### ISSUE THREAD

**Issue by Sethuraman** [realm/SwiftLint#1928]
In the below example, we have a reference to the imagesCollectionView passed into the presenter from the View Controller. The View Controller has a reference to the collection view from the storyboard.

In the presenter we are creating a new ImagesDelegate and assigning to the collection view. If we mark the ImagesDelegate as weak then it will get garbage collected and the collection view will not have a delegate assigned.

The presenter will need to hold a strong reference to the ImagesDelegate. 

This is a false positive reported by the weak_delegate rule.
```
class ImagesPresenter {
    private weak var imagesCollectionView: UICollectionView?
    private var imagesDataSource: ImagesDataSource
    //swiftlint:disable weak_delegate
    private var imagesDelegate: ImagesDelegate
    //swiftlint:enable weak_delegate

    init(viewController: ImagesViewController) {
        self.imagesCollectionView = viewController.imagesCollectionView
        
        imagesDataSource = ImagesDataSource(images: [], imageCount: 0)
        imagesDelegate = ImagesDelegate()
        if let imagesCollectionView = self.imagesCollectionView {
            imagesCollectionView.dataSource = imagesDataSource
            imagesCollectionView.delegate = imagesDelegate
        }
    }
}
```

**Comment by marcelofabri** [comment]
SwiftLint has no way to tell whether a property with a delegate needs to be `strong` or not. It's just common that most times you're holding a reference to a protocol.

Renaming the property, the class or even disabling the rule locally are some options to silence the warning.

**Comment by marcelofabri** [comment]
Closing this due to lack of activity.

**Comment by jeremycochoy** [comment]
Unfortunately, this is still an issue today... 🙁
